src/query/api/v1/handler/database/create.go

```diff
@@ -375,82 +378,185 @@ func defaultedNamespaceAddRequest(
+			blockSize = blockSizeFloor
 		} else {
-			value, err := time.ParseDuration(r.RetentionTime)
-			if err != nil {
-				return nil, fmt.Errorf("invalid retention time: %v", err)
-			}
-			retentionOpts = retentionOpts.SetRetentionPeriod(value)
+			blockSize = blockSizeCeil
 		}
 
-		retentionPeriod := retentionOpts.RetentionPeriod()
+		if blockSize < minRecommendCalculateBlockSize {
+			blockSize = minRecommendCalculateBlockSize
+		} else if blockSize > maxRecommendCalculateBlockSize {
+			blockSize = maxRecommendCalculateBlockSize
+		}
 
-		var blockSize time.Duration
-		switch {
-		case r.BlockSize != nil && r.BlockSize.Time != "":
-			value, err := time.ParseDuration(r.BlockSize.Time)
-			if err != nil {
-				return nil, fmt.Errorf("invalid block size time: %v", err)
-			}
-			blockSize = value
-
-		case r.BlockSize != nil && r.BlockSize.ExpectedSeriesDatapointsPerHour > 0:
-			value := r.BlockSize.ExpectedSeriesDatapointsPerHour
-			blockSize = time.Duration(blockSizeFromExpectedSeriesScalar / value)
-			// Snap to the nearest 5mins
-			blockSizeCeil := blockSize.Truncate(5*time.Minute) + 5*time.Minute
-			blockSizeFloor := blockSize.Truncate(5 * time.Minute)
-			if blockSizeFloor%time.Hour == 0 ||
-				blockSizeFloor%30*time.Minute == 0 ||
-				blockSizeFloor%15*time.Minute == 0 ||
-				blockSizeFloor%10*time.Minute == 0 {
-				// Try snap to hour or 30min or 15min or 10min boundary if possible
-				blockSize = blockSizeFloor
-			} else {
-				blockSize = blockSizeCeil
+	default:
+		// Use the maximum block size if we don't find a way to
+		// recommended one based on request parameters
+		max := recommendedBlockSizesByRetentionAsc[len(recommendedBlockSizesByRetentionAsc)-1]
+		blockSize = max.blockSize
+		for _, elem := range recommendedBlockSizesByRetentionAsc {
+			if retentionPeriod <= elem.forRetentionLessThanOrEqual {
+				blockSize = elem.blockSize
+				break
 			}
+		}
+	}
 
-			if blockSize < minRecommendCalculateBlockSize {
-				blockSize = minRecommendCalculateBlockSize
-			} else if blockSize > maxRecommendCalculateBlockSize {
-				blockSize = maxRecommendCalculateBlockSize
-			}
+	retentionOpts = retentionOpts.SetBlockSize(blockSize)
 
-		default:
-			// Use the maximum block size if we don't find a way to
-			// recommended one based on request parameters
-			max := recommendedBlockSizesByRetentionAsc[len(recommendedBlockSizesByRetentionAsc)-1]
-			blockSize = max.blockSize
-			for _, elem := range recommendedBlockSizesByRetentionAsc {
-				if retentionPeriod <= elem.forRetentionLessThanOrEqual {
-					blockSize = elem.blockSize
-					break
-				}
-			}
+	indexOpts := opts.IndexOptions().
+		SetEnabled(true).
+		SetBlockSize(blockSize)
+
+	opts = opts.SetRetentionOptions(retentionOpts).
+		SetIndexOptions(indexOpts)
+
+	// Resolution does not apply to unaggregated namespaces so set to 0.
+	attrs, err := dbnamespace.NewAggregationAttributes(0, dbnamespace.Unaggregated)
```

Comment:
May be worth adding type-specific constructors (i.e. aggregated which takes in resolution, unaggregated which takes in nothing, and the `dbnamespace.Unaggregated/Aggregated` type is set internally)